great_expectations/data_context/store/validation_config_store.py

```diff
@@ -39,3 +64,14 @@ def serialize(self, value):
     @override
     def deserialize(self, value):
         return ValidationConfig.parse_raw(value)
+
+    @override
+    def _add(self, key, value, **kwargs):
+        if not self.cloud_mode:
+            # this logic should move to the store backend, but is implemented here for now
+            value = self._add_id_on_create(value)
+        return super()._add(key=key, value=value, **kwargs)
+
+    def _add_id_on_create(self, value):
```

Comment:
nit: methods that mutate objects are typically harder to reason about. Any reason not to just have this be `_generate_id(self) -> str`?